json-rpc/src/util.rs

```diff
@@ -177,3 +181,33 @@ pub fn script_view_from_script(script: &Script) -> ScriptView {
 
     view
 }
+
+pub fn script_view_from_script_function(script: &ScriptFunction) -> ScriptView {
+    let name = format!(
+        "{}::{}::{}",
+        script.module().address().short_str_lossless(),
+        script.module().name(),
+        script.function()
+    );
+    let ty_args: Vec<String> = script
+        .ty_args()
+        .iter()
+        .map(|type_tag| match type_tag {
+            TypeTag::Struct(StructTag { module, .. }) => module.to_string(),
+            tag => format!("{}", tag),
+        })
+        .collect();
+    ScriptView {
+        r#type: name,
```

Comment:
This is sort of unfortunate. Really "type" should be "ScriptFun" and there should be another field for the name of the script, but it seems like this is not the case for the older why the ScriptView was constructed.

Generally I look at a JSON field named `type` to be a way to construct a tagged enum so you know how to understand the rest of the payload.